snuba/datasets/entities/transactions.py

```diff
@@ -86,20 +98,89 @@
 )
 
 
+class SampledSimplePipelineBuilder(SimplePipelineBuilder):
+    def build_planner(
+        self, query: LogicalQuery, settings: RequestSettings,
+    ) -> EntityQueryPlanner:
+        new_query = deepcopy(query)
+        new_query.set_sample(0.1)
+        return super().build_planner(new_query, settings)
+
+
 class BaseTransactionsEntity(Entity, ABC):
     def __init__(self, custom_mappers: Optional[TranslationMappers] = None) -> None:
         storage = get_writable_storage(StorageKey.TRANSACTIONS)
         schema = storage.get_table_writer().get_schema()
 
+        pipeline_builder = SimplePipelineBuilder(
+            query_plan_builder=SingleStorageQueryPlanBuilder(
+                storage=storage,
+                mappers=transaction_translator
+                if custom_mappers is None
+                else transaction_translator.concat(custom_mappers),
+            ),
+        )
+        sampled_pipeline_builder = SampledSimplePipelineBuilder(
+            query_plan_builder=SingleStorageQueryPlanBuilder(
+                storage=storage,
+                mappers=transaction_translator
+                if custom_mappers is None
+                else transaction_translator.concat(custom_mappers),
+            ),
+        )
+
+        def selector_func(query: LogicalQuery, referrer: str) -> Tuple[str, List[str]]:
+            if referrer == "tagstore.__get_tag_keys":
```

Comment:
I would also restrict this to project 1 (or a few other big projects if you want). On small projects the sampling will just introduce noise as the sample will not be representative. That would just skew your results.